packages/esm-patient-medications-app/src/api/api.ts

```diff
@@ -51,8 +61,31 @@ export function getDrugByName(drugName: string, abortController?: AbortControlle
   );
 }
 
-export function getDurationUnits(abortController: AbortController, durationUnitsConcept) {
-  return openmrsFetch(`/ws/rest/v1/concept/${durationUnitsConcept}?v=custom:(answers:(uuid,display))`, abortController);
+export function useDurationUnits(durationUnitsConcept) {
+  const { data, error } = useSWRImmutable<FetchResponse<{ answers: Array<OpenmrsResource> }>, Error>(
+    `/ws/rest/v1/concept/${durationUnitsConcept}?v=custom:(answers:(uuid,display))`,
+    openmrsFetch,
+  );
+
+  useEffect(() => {
```

Comment:
Actually, the error comes up in the notification, since there's no error message handling in the UI.

The reason for wrapping in the useEffect is that on re-rendering. and if the error object is still there, it'll show the notification again and again.

Reply:
This error is if the api throws an error, instead of data error